### PR TITLE
Improve performance of large file lists in backup/restore commands.

### PR DESCRIPTION
**Pull request by dwsteele**
lstRemoveIdx(list, 0) resulted in the entire list being moved down to the first position which could take a long time for big lists. This is a common pattern in backup/restore when processing file queues.

Instead simply move the list pointer up when first item is removed. Then on insert check if there is space at the beginning when there is no longer space at the end and do the move then. This way if a list is built and then drained without any new inserts then no move is required.

### CODE REVIEW

File: src/common/type/list.c [hunk 1]
```diff
@@ -294,9 +295,19 @@ lstInsert(List *this, unsigned int listIdx, const void *item)
...
+    // Else if there is space before the beginning of the list then move the list down
+    else if (
+        (this->list != this->listAlloc) &&
+        (this->listSize + ((uintptr_t)(this->list - this->listAlloc) / this->itemSize) == this->listSizeMax))
+    {
+        memmove(this->listAlloc, this->list, this->listSize * this->itemSize);
+        this->list = this->listAlloc;
+    }
 
     // If not inserting at the end then move items down to make space
```
Comment (cmwshang) [inline src/common/type/list.c:312]: Could you move this comment  `// If not inserting at the end then move items down to make space` to right before the `if (listIdx != lstSize(this))` statement?

**Review by cmwshang — COMMENTED** [review]
Looks good. One minor request to relocate a comment...

File: src/common/type/list.c (hunk 1 above)
Comment (dwsteele) [reply src/common/type/list.c]: Sure. Also added a comment.

Comment (cmwshang) [reply src/common/type/list.c]: Awesome.

**Review by cmwshang — APPROVED** [review]

**Pull request merged**